halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/SpringProfileFactory.java

```diff
@@ -41,6 +43,24 @@ protected void setProfile(
 
     profile.appendContents(
         yamlToString(deploymentConfiguration.getName(), profile, spectatorConfig));
+
+    if (addExtensibilityConfigs()) {
+      Map<String, Object> spinnakerYaml = new LinkedHashMap<>();
+      Map<String, Object> extensibilityYaml = new LinkedHashMap<>();
+      Map<String, Object> extensibilityContents =
+          deploymentConfiguration.getSpinnaker().getExtensibility().toMap();
+      extensibilityContents.put(
+          "plugins-root-path", "/opt/" + this.getArtifact().toString().toLowerCase() + "/plugins");
+      extensibilityYaml.put("extensibility", extensibilityContents);
+      spinnakerYaml.put("spinnaker", extensibilityYaml);
+
+      profile.appendContents(
+          yamlToString(deploymentConfiguration.getName(), profile, spinnakerYaml));
+    }
+  }
+
+  public Boolean addExtensibilityConfigs() {
```

Comment:
I think this can be `protected`.  It should also return a `boolean` instead of a `Boolean` because we never expect it to be null.